src/spatial/collision.js

```diff
@@ -118,7 +118,7 @@ Crafty.c("Collision", {
         }
 
         // Finally, assign the hitbox, and attach it to the "Collision" entity
-        this.map = poly;
+        this.map = poly.clone();
```

Comment:
This would be the wrong place to clone.  It should probably be in an else branch on line 102, so that the logic is as follows:

```
if (arguments.length > 1 ) { 
   // create polygon from arguments
} else {
   // clone polygon from single argument
}
```